RoaringBitmap/src/main/java/org/roaringbitmap/BitSetUtil.java

```diff
@@ -1,24 +1,43 @@
 package org.roaringbitmap;
 
 
+import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
 import java.util.BitSet;
 
 
 /***
- *
  * This class provides convenience functions to manipulate BitSet and RoaringBitmap objects.
  *
  */
 public class BitSetUtil {
-  // todo: add a method to convert a RoaringBitmap to a BitSet using BitSet.valueOf
-
   // a block consists has a maximum of 1024 words, each representing 64 bits,
   // thus representing at maximum 65536 bits
-  public static final int BLOCK_LENGTH = BitmapContainer.MAX_CAPACITY / Long.SIZE; //
-  // 64-bit
-  // word
+  public static final int BLOCK_LENGTH = BitmapContainer.MAX_CAPACITY / Long.SIZE;
+
+  /**
+   * Converts an immutable roaring bitmap to JDK bit set.
+   *
```

Comment:
I have created [some benchmark](https://github.com/RoaringBitmap/RoaringBitmap/commit/da1dead0ce8bb4bc7222869ca5c37c6b477edd1f) to optimize conversion. There is trade off between too much code and performance. **What variant do you suggest to choose? Should I create separate PR for this or it can be included here?**

| Benchmark | inputData | Mode | Cnt | Score | Error | Units |
| --- | --- | --- | --- | --- | --- | --- |
| iterateThroughContainers | CONSECUTIVE | avgt | 5 | 52.782 | ±3.963 | ms/op |
| iterateThroughContainers | RANDOM | avgt | 5 | 66.945 | ±8.262 | ms/op |
| negativeValuesSeparately | CONSECUTIVE | avgt | 5 | 62.152 | ±1.104 | ms/op |
| negativeValuesSeparately | RANDOM | avgt | 5 | 85.505 | ±12.954 | ms/op |
| negativeValuesSeparatelyRegisterPreviousValue | CONSECUTIVE| avgt | 5 | 49.961 | ±8.429 | ms/op |
| negativeValuesSeparatelyRegisterPreviousValue| RANDOM | avgt |5 | 185.567 | ±5.139 | ms/op |
| preallocated | CONSECUTIVE | avgt | 5 | 67.221 | ±1.658 | ms/op |
| preallocated | RANDOM | avgt | 5 | 78.673 | ±5.566 | ms/op |
| original | CONSECUTIVE | avgt | 5 | 75.124 | ±27.983 | ms/op |
| original | RANDOM | avgt | 5 | 84.717 | ±3.251 | ms/op |

Reply:
The variant using streaming API inspired by #556 was added afterwards and it is magically the fastest for consecutive values.

Reply:
> There is trade off between too much code and performance. What variant do you suggest to choose?

My concern here was that the library user should know what to expect. If we provide a convenience function that is possibly slow, we should warn them... otherwise it can be considered a performance bug.

Not everything needs to be super optimized with complex code, but the user should get some warning if the performance can be poor.

Reply:
And what variant you suggest we should use? As I wrote earlier, I have created [benchmarks](https://github.com/RoaringBitmap/RoaringBitmap/commit/da1dead0ce8bb4bc7222869ca5c37c6b477edd1f) for this issue, where are already optimized implementation. Why I ask you what to choose? If you look on benchmark results, there is no solution dominating both for tested cases - for consecutive values is the best variant using _streamingApi_ and for random values _iterateThroughContainers_. After that I can merge chosen one into this PR and do some code clean up - at least comments as they are wrong if I remember right.

Reply:
Ok. The issue is that your implementation is fine, but you do bit-by-bit conversion.
If the RoaringBitmap contains a bitset container, for example. Then you can just use [valueOf](https://docs.oracle.com/javase/8/docs/api/java/util/BitSet.html#valueOf-long:A-)(long[] longs) to map directly, at a highspeed, the content of the Roaring container to a JDK BitSet. You are obviously aware of this because you have it in a comment.

I don't think we even need benchmarks, it is going to be much faster... 

Now. It is fine *not* implementing this, but I would request that you then document the performance in the document of the function. (E.g., "this function may set the bits one by one and may be computationally intensive".) That's important because some users might decide to systematically convert their Roaring bitmaps to BitSet. We need to warn them about the performance implications.

The Go roaring implementation and the C Roaring implementations have equivalent functionality btw.

Reply:
Here is the C implementation: https://github.com/RoaringBitmap/CRoaring/blob/daad140f013cbb4e8b515da264403843d8dc1755/src/roaring.c#L3306

Reply:
Here is the Go implementation:
https://github.com/RoaringBitmap/roaring/blob/88e9207dc61ee814891a2a7cce4aa8abf1fbac1f/roaring.go#L186

Reply:
I am *not* asking you to use these sorts of implementations. They require more testing and more effort. I am just asking you to send the users a signal regarding how it is currently implemented.